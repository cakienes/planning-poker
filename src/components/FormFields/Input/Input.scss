.defaultInputHolder {
    margin-top: 40px;
    .defaultInput {
        width: 100%;
        height: 66px;
        padding: 20px 10px;
        font-size: 24px;
        align-items: center;
        outline: none;

        &::placeholder {
            color: #ddd;
        }
    }

    .defaultLabel {
        padding: 0 10px;
        font-size: 22px;
    }

    .error {
        border: 1px solid red;
    }
}
