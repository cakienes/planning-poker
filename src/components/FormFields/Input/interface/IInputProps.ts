import { WrappedFieldProps } from 'redux-form';

export default interface IInputProps extends WrappedFieldProps {
    label?: string;
    placeholder?: string;
    type?: string;
}
