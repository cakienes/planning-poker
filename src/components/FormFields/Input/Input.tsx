import React from 'react';
import './Input.scss';
import IInputProps from './interface/IInputProps';

const Input = (props: IInputProps) => {
    const {
        input,
        meta: { touched, error },
        type,
        placeholder,
    } = props;
    return (
        <div className="defaultInputHolder">
            <label className="defaultLabel">{props.label}</label>
            <div className={`${touched && error ? 'error' : undefined}`}>
                <input id={input.name} className="defaultInput" type={type} placeholder={placeholder} {...input} />
            </div>
        </div>
    );
};

export default Input;
