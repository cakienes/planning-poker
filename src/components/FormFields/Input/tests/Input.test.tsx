import { shallow } from 'enzyme';
import React from 'react';
import Input from '../Input';
import IInputProps from '../interface/IInputProps';

describe('Header.tsx', () => {
    const inputProps: IInputProps = {
        label: 'label',
        name: 'name',
        placeholder: 'placeholder',
    };
    it('renders without crashing', async () => {
        expect(shallow(<Input {...inputProps} />)).toMatchSnapshot();
    });
});
