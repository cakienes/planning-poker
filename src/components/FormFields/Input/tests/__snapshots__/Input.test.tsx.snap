// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.tsx renders without crashing 1`] = `ShallowWrapper {}`;
